src/Symfony/Bundle/TwigBundle/Resources/views/Exception/traces.html.twig

```diff
@@ -22,10 +22,11 @@
         <div id="trace-html-{{ index }}" class="sf-toggle-content">
         {% set _is_first_user_code = true %}
         {% for i, trace in exception.trace %}
-            {% set _display_code_snippet = _is_first_user_code and ('/vendor/' not in trace.file) and ('/var/cache/' not in trace.file) and (trace.file is not empty) %}
+            {% set _is_vendor_trace = trace.file is not empty and ('/vendor/' in trace.file or '/var/cache/' in trace.file) %}
+            {% set _display_code_snippet = _is_first_user_code and not _is_vendor_trace %}
             {% if _display_code_snippet %}{% set _is_first_user_code = false %}{% endif %}
-            <div class="trace-line">
-                {{ include('@Twig/Exception/trace.html.twig', { prefix: index, i: i, trace: trace, _display_code_snippet: _display_code_snippet }, with_context = false) }}
+            <div class="trace-line {{ _is_vendor_trace ? 'trace-from-vendor' }}">
+                {{ include('@Twig/Exception/trace.html.twig', { prefix: index, i: i, trace: trace, style: _is_vendor_trace ? 'compact' : _display_code_snippet ? 'expanded' }, with_context = false) }}
```

Comment:
Shouldn't it be `style: _is_vendor_trace ? 'compact' : 'expanded'` instead? 🤔

Reply:
This is a bit trickier because we have 3 states: `compact` for vendors, `expanded` for the only trace that displays the code snippet expanded (there should be just one) and `normal` which displays your code but doesn't expand the code snippet.

Reply:
Maybe I'm missing something, but doesn't this check will interpret any application path containing "/vendor/" as part of vendor code? I know my point is really border, but could we check if the path is really in the project's `/vendor` directory?

Reply:
Although you are right, I'd prefer to ignore that edge case: having `/vendor/` in the file path and not being a real vendor. Moreover, in the last iteration of this feature we no longer hide anything, so the user won't miss anything important even in that edge case. Cheers!